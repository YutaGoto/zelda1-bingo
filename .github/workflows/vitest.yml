name: vitest

on:
  - push

jobs:
  vitest:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout your repository using git
        uses: actions/checkout@v4
      - name: Setup pnpm
        uses: pnpm/action-setup@v2
        with:
          version: 8
      - name: Setup Node.js
        uses: actions/setup-node@v4
        with:
          cache: "pnpm"
          node-version: 20
      - name: Install dependencies
        run: pnpm install
      - name: Run tests
        run: pnpm test
