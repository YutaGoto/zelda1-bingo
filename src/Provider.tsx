import {
  ChakraProvider,
  ColorModeScript,
  type ThemeConfig,
  extendTheme,
} from "@chakra-ui/react";
import { type StyleFunctionProps, mode } from "@chakra-ui/theme-tools";
import type { ReactNode } from "react";

interface ProviderProps {
  children: ReactNode;
}

export const Provider = ({ children }: ProviderProps) => {
  const config: ThemeConfig = {
    initialColorMode: "system",
    useSystemColorMode: true,
  };

  const theme = extendTheme({
    ...config,
    styles: {
      global: (props: StyleFunctionProps) => ({
        body: {
          bg: mode("gray.50", "gray.800")(props),
        },
      }),
    },
    fonts: {
      heading: `'Noto Sans JP Variable', sans-serif`,
      body: `'Noto Sans JP Variable', sans-serif`,
    },
  });

  return (
    <ChakraProvider theme={theme}>
      <ColorModeScript initialColorMode={config.initialColorMode} />
      {children}
    </ChakraProvider>
  );
};
