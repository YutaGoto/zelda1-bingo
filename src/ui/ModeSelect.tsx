import { Box, Button, Grid, Link, Text } from "@chakra-ui/react";
import { t } from "i18next";

interface ModeSelectProps {
  mode: "score" | "bingo" | "single";
  category: "firstQuest" | "secondQuest" | "swordless";
  lang: string | undefined;
  mt: number;
}

export const ModeSelect = ({ mode, lang, category, mt }: ModeSelectProps) => {
  return (
    <Box mt={mt}>
      <Text fontSize="lg">{t("mode")}</Text>
      <Grid
        templateColumns={{ lg: "repeat(2, 1fr)", md: "repeat(4, 1fr)" }}
        gap={2}
        mt={2}
      >
        <Button
          as={Link}
          href={`/${category}/${lang}`}
          variant={mode === "bingo" ? "solid" : "outline"}
          colorScheme="cyan"
        >
          {t("bingo")}
        </Button>
        <Button
          as={Link}
          href={`/score/${category}/${lang}`}
          colorScheme="yellow"
          variant={mode === "score" ? "solid" : "outline"}
        >
          {t("score")}
        </Button>
        <Button
          // as={Link}
          isDisabled={true}
          // href={`/single/${category}/${lang}`}
          colorScheme="red"
          variant={mode === "single" ? "solid" : "outline"}
        >
          {t("singleTasking")}
        </Button>
      </Grid>
    </Box>
  );
};
