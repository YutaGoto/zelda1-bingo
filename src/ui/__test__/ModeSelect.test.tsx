import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import { Provider } from "../../Provider";
import { ModeSelect } from "../ModeSelect";

describe("Mode select languages", () => {
  test("title is English", () => {
    render(<ModeSelect mode="bingo" category="firstQuest" lang="en" />, {
      wrapper: Provider,
    });

    expect(screen.getByText("Mode")).toBeTruthy();
  });
});

describe("active mode", () => {
  test("bingo", () => {
    const { asFragment } = render(
      <ModeSelect mode="bingo" category="swordless" lang="en" />,
      { wrapper: Provider },
    );

    expect(screen.getByText("Bingo")).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  test("score attack", () => {
    const { asFragment } = render(
      <ModeSelect mode="score" category="firstQuest" lang="en" />,
      { wrapper: Provider },
    );

    expect(screen.getByText("Score Attack")).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  // test("single tasking", () => {
  //   render(<ModeSelect mode="tasking" category="firstQuest" lang="en" />);

  //   expect(screen.getByText("1st Quest")).toBeTruthy();
  // });
});
