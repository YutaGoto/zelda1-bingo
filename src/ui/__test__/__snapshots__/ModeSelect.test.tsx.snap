// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`active mode > bingo 1`] = `
<DocumentFragment>
  <div
    class="css-zv7ju9"
  >
    <p
      class="chakra-text css-1ezsviu"
    >
      Mode
    </p>
    <div
      class="css-k5oabe"
    >
      <a
        class="chakra-link chakra-button css-1ncwzw6"
        href="/swordless/en"
      >
        Bingo
      </a>
      <a
        class="chakra-link chakra-button css-1ncwzw6"
        href="/score/swordless/en"
      >
        Score Attack
      </a>
      <button
        class="chakra-button css-4xx2wk"
        disabled=""
        type="button"
      >
        Single Tasking
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`active mode > score attack 1`] = `
<DocumentFragment>
  <div
    class="css-zv7ju9"
  >
    <p
      class="chakra-text css-1ezsviu"
    >
      Mode
    </p>
    <div
      class="css-k5oabe"
    >
      <a
        class="chakra-link chakra-button css-1ncwzw6"
        href="/firstQuest/en"
      >
        Bingo
      </a>
      <a
        class="chakra-link chakra-button css-1ncwzw6"
        href="/score/firstQuest/en"
      >
        Score Attack
      </a>
      <button
        class="chakra-button css-4xx2wk"
        disabled=""
        type="button"
      >
        Single Tasking
      </button>
    </div>
  </div>
</DocumentFragment>
`;
